src/keystore.h

```diff
@@ -102,6 +115,16 @@ class CBasicKeyStore : public CKeyStore
     virtual bool HaveCScript(const CScriptID &hash) const;
     virtual bool GetCScript(const CScriptID &hash, CScript& redeemScriptOut) const;
 
+    bool GetPreimage(
+        const std::vector<unsigned char>& image,
+        std::vector<unsigned char>& preimage
+    ) const;
+    
+    bool AddPreimage(
+        const std::vector<unsigned char>& image,
+        const std::vector<unsigned char>& preimage
+    );
+
```

Comment:
Maybe add a RemovePreimage, LockPreimage, and UnlockPreimage call, which could be used to ensure that a node no longer has the ability to solve a particular HTLC or does not until explicitly allowed. These should also go to disk.